gateway-js/src/__tests__/executeQueryPlan.test.ts

```diff
@@ -3516,4 +3516,1113 @@ describe('executeQueryPlan', () => {
         `);
     });
   });
```

Comment:
You talk in the comments about potentially having a naming collision when it comes to aliasing and potentially needing to regenerate the name? Is there a test for that case?

Reply:
Good call. I had not added a test for that, so pushed one (which obviously uncovered a small issue in some cases).